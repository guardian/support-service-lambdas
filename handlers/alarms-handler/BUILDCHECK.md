# Buildcheck managed file list
The files listed below are managed by buildcheck and their content is checked by the build.

## HOWTO edit managed files
1. edit the build definition in buildcheck/data/
2. run `pnpm update-build` at the root

For further details, see [buildcheck/README.md](../../buildcheck/README.md)

## Generated file list:
- [cdk/.prettierignore](cdk/.prettierignore)
- [cdk/cdk.json](cdk/cdk.json)
- [cdk/jest.setup.js](cdk/jest.setup.js)
- [cdk/package.json](cdk/package.json)
- [cdk/tsconfig.json](cdk/tsconfig.json)
- [lambda/jest.config.js](lambda/jest.config.js)
- [lambda/package.json](lambda/package.json)
- [lambda/tsconfig.json](lambda/tsconfig.json)
- [riff-raff.yaml](riff-raff.yaml)
- [BUILDCHECK.md](BUILDCHECK.md)

This file is generated by [buildcheck/data/snippets/BUILDCHECK.md.ts](../../buildcheck/data/snippets/BUILDCHECK.md.ts)
