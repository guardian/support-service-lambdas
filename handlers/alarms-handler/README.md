# Alarms Handler

## Landscape Diagram

![landscape diagram](./docs/landscape-diagram.png 'Landscape Diagram')

## Context Diagram

![landscape diagram](./docs/container-diagram.png 'Container Diagram')
