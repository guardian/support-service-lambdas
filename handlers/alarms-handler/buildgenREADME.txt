#
# The files listed below are managed by buildgen.
#
# To make changes without failing the build:
# 1. edit the build definition in buildgen/src/data/
# 2. run `pnpm buildgen` at the root
#
# For further details, see buildgen/README.md
#
# Generated file list:
#
cdk/.prettierignore
cdk/cdk.json
cdk/jest.setup.js
cdk/package.json
cdk/tsconfig.json
lambda/jest.config.js
lambda/package.json
lambda/tsconfig.json
riff-raff.yaml
buildgenREADME.txt
#
# This file is generated by buildgen/src/data/snippets/buildgenREADME.txt.ts