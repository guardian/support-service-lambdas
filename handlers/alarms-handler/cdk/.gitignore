cdk.out
.prettierignore
cdk.json
jest.setup.js
package.json
tsconfig.json
