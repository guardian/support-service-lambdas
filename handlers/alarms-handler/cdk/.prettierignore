# GENERATED FILE: see buildcheck/README.md - template: buildcheck/data/handlerTemplate/cdk/.prettierignore.ts
dist/
