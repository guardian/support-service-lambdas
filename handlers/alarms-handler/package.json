{
	"name": "alarms-handler",
	"scripts": {
		"test": "jest --group=-integration",
		"it-test": "jest --group=integration",
		"type-check": "tsc --noEmit",
		"build": "esbuild --bundle --platform=node --target=node18 --outfile=target/index.js src/index.ts",
		"lint": "eslint src/**/*.ts",
		"package": "pnpm type-check && pnpm lint && pnpm test && pnpm build && cd target; zip -qr alarms-handler.zip ./*.js"
	},
	"dependencies": {
		"@aws-sdk/client-cloudwatch": "^3.592.0",
		"@aws-sdk/credential-providers": "^3.556.0",
		"zod": "^3.23.4"
	},
	"devDependencies": {
		"@types/aws-lambda": "^8.10.132"
	}
}
