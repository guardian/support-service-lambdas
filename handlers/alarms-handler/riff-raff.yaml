# MANAGED FILE: to push changes see buildcheck/README.md - template: buildcheck/data/templates/handler/riff-raff.yaml.ts
stacks:
  - support
regions:
  - eu-west-1
allowedStages:
  - CODE
  - PROD
deployments:
  alarms-handler-cloudformation:
    type: cloud-formation
    app: alarms-handler
    parameters:
      templateStagePaths:
        CODE: alarms-handler-CODE.template.json
        PROD: alarms-handler-PROD.template.json
  alarms-handler:
    type: aws-lambda
    parameters:
      fileName: alarms-handler.zip
      bucketSsmLookup: true
      prefixStack: false
      functionNames:
        - alarms-handler-
        - alarms-handler-scheduled-
    dependencies:
      - alarms-handler-cloudformation
