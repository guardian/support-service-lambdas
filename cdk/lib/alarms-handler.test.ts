import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { AlarmsHandler } from './alarms-handler';

describe('The alarms-handler stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const codeStack = new AlarmsHandler(app, 'alarms-handler-CODE', {
			stack: 'membership',
			stage: 'CODE',
		});
		const prodStack = new AlarmsHandler(app, 'alarms-handler-PROD', {
			stack: 'membership',
			stage: 'PROD',
		});

		expect(Template.fromStack(codeStack).toJSON()).toMatchSnapshot();
		expect(Template.fromStack(prodStack).toJSON()).toMatchSnapshot();
	});
});
