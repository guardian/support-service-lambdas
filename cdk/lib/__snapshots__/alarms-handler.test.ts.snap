// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The alarms-handler stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuDistributionBucketParameter",
      "GuLambdaFunction",
      "GuAlarm",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "alarmshandlerGROWTHwebhook": {
      "Description": "GROWTH Team Google Chat webhook URL",
      "Type": "String",
    },
    "alarmshandlerPPwebhook": {
      "Description": "PP Team Google Chat webhook URL",
      "Type": "String",
    },
    "alarmshandlerSREwebhook": {
      "Description": "SRE Team Google Chat webhook URL",
      "Type": "String",
    },
    "alarmshandlerVALUEwebhook": {
      "Description": "VALUE Team Google Chat webhook URL",
      "Type": "String",
    },
    "alarmshandlermobileaccountrolearn": {
      "Description": "ARN of role in the mobile account which allows cloudwatch:ListTagsForResource",
      "Type": "String",
    },
    "alarmshandlermobileawsaccount": {
      "Description": "ID of the mobile aws account",
      "Type": "String",
    },
    "alarmshandlertargetingawsaccount": {
      "Description": "ID of the targeting aws account",
      "Type": "String",
    },
  },
  "Resources": {
    "alarmshandleralarm5F97F469": {
      "Properties": {
        "ActionsEnabled": false,
        "AlarmActions": [
          {
            "Fn::Join": [
              "",
              [
                "arn:aws:sns:",
                {
                  "Ref": "AWS::Region",
                },
                ":",
                {
                  "Ref": "AWS::AccountId",
                },
                ":",
                {
                  "Fn::GetAtt": [
                    "alarmshandlertopic003C974E",
                    "TopicName",
                  ],
                },
              ],
            ],
          },
        ],
        "AlarmDescription": "There was an error in the lambda function that handles CloudWatch alarms.",
        "AlarmName": "CODE: Failed to handle CloudWatch alarm",
        "ComparisonOperator": "GreaterThanThreshold",
        "Dimensions": [
          {
            "Name": "QueueName",
            "Value": {
              "Fn::GetAtt": [
                "deadlettersalarmshandlerqueueC7A67616",
                "QueueName",
              ],
            },
          },
        ],
        "EvaluationPeriods": 24,
        "MetricName": "ApproximateNumberOfMessagesVisible",
        "Namespace": "AWS/SQS",
        "Period": 60,
        "Statistic": "Sum",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "alarmshandlercloudwatchpolicy7746710B": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "cloudwatch:ListTagsForResource",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "alarmshandlercloudwatchpolicy7746710B",
        "Roles": [
          {
            "Ref": "alarmshandlerlambdaServiceRole3953DF7F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "alarmshandlerlambda383D2BAA": {
      "DependsOn": [
        "alarmshandlerlambdaServiceRoleDefaultPolicyFEF447F6",
        "alarmshandlerlambdaServiceRole3953DF7F",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "membership/CODE/alarms-handler/alarms-handler.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "alarms-handler",
            "GROWTH_WEBHOOK": {
              "Ref": "alarmshandlerGROWTHwebhook",
            },
            "MOBILE_AWS_ACCOUNT_ID": {
              "Ref": "alarmshandlermobileawsaccount",
            },
            "MOBILE_ROLE_ARN": {
              "Ref": "alarmshandlermobileaccountrolearn",
            },
            "PP_WEBHOOK": {
              "Ref": "alarmshandlerPPwebhook",
            },
            "SRE_WEBHOOK": {
              "Ref": "alarmshandlerSREwebhook",
            },
            "STACK": "membership",
            "STAGE": "CODE",
            "VALUE_WEBHOOK": {
              "Ref": "alarmshandlerVALUEwebhook",
            },
          },
        },
        "FunctionName": "alarms-handler-CODE",
        "Handler": "index.handler",
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "alarmshandlerlambdaServiceRole3953DF7F",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "alarms-handler",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
        "Timeout": 15,
      },
      "Type": "AWS::Lambda::Function",
    },
    "alarmshandlerlambdaServiceRole3953DF7F": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "alarms-handler",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "alarmshandlerlambdaServiceRoleDefaultPolicyFEF447F6": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "sqs:ReceiveMessage",
                "sqs:ChangeMessageVisibility",
                "sqs:GetQueueUrl",
                "sqs:DeleteMessage",
                "sqs:GetQueueAttributes",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "alarmshandlerqueue255509D6",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/membership/CODE/alarms-handler/alarms-handler.zip",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/CODE/membership/alarms-handler",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/CODE/membership/alarms-handler/*",
                  ],
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Ref": "alarmshandlermobileaccountrolearn",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "alarmshandlerlambdaServiceRoleDefaultPolicyFEF447F6",
        "Roles": [
          {
            "Ref": "alarmshandlerlambdaServiceRole3953DF7F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "alarmshandlerlambdaSqsEventSourcealarmshandlerCODEalarmshandlerqueue16858B0A6ACA8CB7": {
      "Properties": {
        "EventSourceArn": {
          "Fn::GetAtt": [
            "alarmshandlerqueue255509D6",
            "Arn",
          ],
        },
        "FunctionName": {
          "Ref": "alarmshandlerlambda383D2BAA",
        },
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
    "alarmshandlerqueue255509D6": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "alarms-handler-queue-CODE",
        "RedrivePolicy": {
          "deadLetterTargetArn": {
            "Fn::GetAtt": [
              "deadlettersalarmshandlerqueueC7A67616",
              "Arn",
            ],
          },
          "maxReceiveCount": 3,
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "alarmshandlerqueuePolicy2451CBBF": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "sqs:SendMessage",
              "Condition": {
                "ArnEquals": {
                  "aws:SourceArn": {
                    "Ref": "alarmshandlertopic003C974E",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "sns.amazonaws.com",
              },
              "Resource": {
                "Fn::GetAtt": [
                  "alarmshandlerqueue255509D6",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Queues": [
          {
            "Ref": "alarmshandlerqueue255509D6",
          },
        ],
      },
      "Type": "AWS::SQS::QueuePolicy",
    },
    "alarmshandlerqueuealarmshandlerCODEalarmshandlertopic53848409AA067E1C": {
      "DependsOn": [
        "alarmshandlerqueuePolicy2451CBBF",
      ],
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "alarmshandlerqueue255509D6",
            "Arn",
          ],
        },
        "Protocol": "sqs",
        "TopicArn": {
          "Ref": "alarmshandlertopic003C974E",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "alarmshandlertopic003C974E": {
      "Properties": {
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
        "TopicName": "alarms-handler-topic-CODE",
      },
      "Type": "AWS::SNS::Topic",
    },
    "alarmshandlertopicPolicy7401CE24": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": [
                    {
                      "Fn::Join": [
                        "",
                        [
                          "arn:aws:cloudwatch:eu-west-1:",
                          {
                            "Ref": "alarmshandlermobileawsaccount",
                          },
                          ":alarm:*",
                        ],
                      ],
                    },
                    {
                      "Fn::Join": [
                        "",
                        [
                          "arn:aws:cloudwatch:eu-west-1:",
                          {
                            "Ref": "alarmshandlertargetingawsaccount",
                          },
                          ":alarm:*",
                        ],
                      ],
                    },
                    {
                      "Fn::Join": [
                        "",
                        [
                          "arn:aws:cloudwatch:eu-west-1:",
                          {
                            "Ref": "AWS::AccountId",
                          },
                          ":alarm:*",
                        ],
                      ],
                    },
                  ],
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "alarmshandlertopic003C974E",
              },
              "Sid": "0",
            },
          ],
          "Version": "2012-10-17",
        },
        "Topics": [
          {
            "Ref": "alarmshandlertopic003C974E",
          },
        ],
      },
      "Type": "AWS::SNS::TopicPolicy",
    },
    "deadlettersalarmshandlerqueueC7A67616": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "MessageRetentionPeriod": 1209600,
        "QueueName": "dead-letters-alarms-handler-queue-CODE",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;

exports[`The alarms-handler stack matches the snapshot 2`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuStringParameter",
      "GuDistributionBucketParameter",
      "GuLambdaFunction",
      "GuAlarm",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "alarmshandlerGROWTHwebhook": {
      "Description": "GROWTH Team Google Chat webhook URL",
      "Type": "String",
    },
    "alarmshandlerPPwebhook": {
      "Description": "PP Team Google Chat webhook URL",
      "Type": "String",
    },
    "alarmshandlerSREwebhook": {
      "Description": "SRE Team Google Chat webhook URL",
      "Type": "String",
    },
    "alarmshandlerVALUEwebhook": {
      "Description": "VALUE Team Google Chat webhook URL",
      "Type": "String",
    },
    "alarmshandlermobileaccountrolearn": {
      "Description": "ARN of role in the mobile account which allows cloudwatch:ListTagsForResource",
      "Type": "String",
    },
    "alarmshandlermobileawsaccount": {
      "Description": "ID of the mobile aws account",
      "Type": "String",
    },
    "alarmshandlertargetingawsaccount": {
      "Description": "ID of the targeting aws account",
      "Type": "String",
    },
  },
  "Resources": {
    "alarmshandleralarm5F97F469": {
      "Properties": {
        "ActionsEnabled": true,
        "AlarmActions": [
          {
            "Fn::Join": [
              "",
              [
                "arn:aws:sns:",
                {
                  "Ref": "AWS::Region",
                },
                ":",
                {
                  "Ref": "AWS::AccountId",
                },
                ":",
                {
                  "Fn::GetAtt": [
                    "alarmshandlertopic003C974E",
                    "TopicName",
                  ],
                },
              ],
            ],
          },
        ],
        "AlarmDescription": "There was an error in the lambda function that handles CloudWatch alarms.",
        "AlarmName": "PROD: Failed to handle CloudWatch alarm",
        "ComparisonOperator": "GreaterThanThreshold",
        "Dimensions": [
          {
            "Name": "QueueName",
            "Value": {
              "Fn::GetAtt": [
                "deadlettersalarmshandlerqueueC7A67616",
                "QueueName",
              ],
            },
          },
        ],
        "EvaluationPeriods": 24,
        "MetricName": "ApproximateNumberOfMessagesVisible",
        "Namespace": "AWS/SQS",
        "Period": 60,
        "Statistic": "Sum",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "alarmshandlercloudwatchpolicy7746710B": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "cloudwatch:ListTagsForResource",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "alarmshandlercloudwatchpolicy7746710B",
        "Roles": [
          {
            "Ref": "alarmshandlerlambdaServiceRole3953DF7F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "alarmshandlerlambda383D2BAA": {
      "DependsOn": [
        "alarmshandlerlambdaServiceRoleDefaultPolicyFEF447F6",
        "alarmshandlerlambdaServiceRole3953DF7F",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "membership/PROD/alarms-handler/alarms-handler.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "alarms-handler",
            "GROWTH_WEBHOOK": {
              "Ref": "alarmshandlerGROWTHwebhook",
            },
            "MOBILE_AWS_ACCOUNT_ID": {
              "Ref": "alarmshandlermobileawsaccount",
            },
            "MOBILE_ROLE_ARN": {
              "Ref": "alarmshandlermobileaccountrolearn",
            },
            "PP_WEBHOOK": {
              "Ref": "alarmshandlerPPwebhook",
            },
            "SRE_WEBHOOK": {
              "Ref": "alarmshandlerSREwebhook",
            },
            "STACK": "membership",
            "STAGE": "PROD",
            "VALUE_WEBHOOK": {
              "Ref": "alarmshandlerVALUEwebhook",
            },
          },
        },
        "FunctionName": "alarms-handler-PROD",
        "Handler": "index.handler",
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "alarmshandlerlambdaServiceRole3953DF7F",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "alarms-handler",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
        "Timeout": 15,
      },
      "Type": "AWS::Lambda::Function",
    },
    "alarmshandlerlambdaServiceRole3953DF7F": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "alarms-handler",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "alarmshandlerlambdaServiceRoleDefaultPolicyFEF447F6": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "sqs:ReceiveMessage",
                "sqs:ChangeMessageVisibility",
                "sqs:GetQueueUrl",
                "sqs:DeleteMessage",
                "sqs:GetQueueAttributes",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "alarmshandlerqueue255509D6",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/membership/PROD/alarms-handler/alarms-handler.zip",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/PROD/membership/alarms-handler",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/PROD/membership/alarms-handler/*",
                  ],
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Ref": "alarmshandlermobileaccountrolearn",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "alarmshandlerlambdaServiceRoleDefaultPolicyFEF447F6",
        "Roles": [
          {
            "Ref": "alarmshandlerlambdaServiceRole3953DF7F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "alarmshandlerlambdaSqsEventSourcealarmshandlerPRODalarmshandlerqueue21D5C3E59ECCFD65": {
      "Properties": {
        "EventSourceArn": {
          "Fn::GetAtt": [
            "alarmshandlerqueue255509D6",
            "Arn",
          ],
        },
        "FunctionName": {
          "Ref": "alarmshandlerlambda383D2BAA",
        },
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
    "alarmshandlerqueue255509D6": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "alarms-handler-queue-PROD",
        "RedrivePolicy": {
          "deadLetterTargetArn": {
            "Fn::GetAtt": [
              "deadlettersalarmshandlerqueueC7A67616",
              "Arn",
            ],
          },
          "maxReceiveCount": 3,
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "alarmshandlerqueuePolicy2451CBBF": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "sqs:SendMessage",
              "Condition": {
                "ArnEquals": {
                  "aws:SourceArn": {
                    "Ref": "alarmshandlertopic003C974E",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "sns.amazonaws.com",
              },
              "Resource": {
                "Fn::GetAtt": [
                  "alarmshandlerqueue255509D6",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Queues": [
          {
            "Ref": "alarmshandlerqueue255509D6",
          },
        ],
      },
      "Type": "AWS::SQS::QueuePolicy",
    },
    "alarmshandlerqueuealarmshandlerPRODalarmshandlertopic952FCC2B727BB750": {
      "DependsOn": [
        "alarmshandlerqueuePolicy2451CBBF",
      ],
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "alarmshandlerqueue255509D6",
            "Arn",
          ],
        },
        "Protocol": "sqs",
        "TopicArn": {
          "Ref": "alarmshandlertopic003C974E",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "alarmshandlertopic003C974E": {
      "Properties": {
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
        "TopicName": "alarms-handler-topic-PROD",
      },
      "Type": "AWS::SNS::Topic",
    },
    "alarmshandlertopicPolicy7401CE24": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": [
                    {
                      "Fn::Join": [
                        "",
                        [
                          "arn:aws:cloudwatch:eu-west-1:",
                          {
                            "Ref": "alarmshandlermobileawsaccount",
                          },
                          ":alarm:*",
                        ],
                      ],
                    },
                    {
                      "Fn::Join": [
                        "",
                        [
                          "arn:aws:cloudwatch:eu-west-1:",
                          {
                            "Ref": "alarmshandlertargetingawsaccount",
                          },
                          ":alarm:*",
                        ],
                      ],
                    },
                    {
                      "Fn::Join": [
                        "",
                        [
                          "arn:aws:cloudwatch:eu-west-1:",
                          {
                            "Ref": "AWS::AccountId",
                          },
                          ":alarm:*",
                        ],
                      ],
                    },
                  ],
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "alarmshandlertopic003C974E",
              },
              "Sid": "0",
            },
          ],
          "Version": "2012-10-17",
        },
        "Topics": [
          {
            "Ref": "alarmshandlertopic003C974E",
          },
        ],
      },
      "Type": "AWS::SNS::TopicPolicy",
    },
    "deadlettersalarmshandlerqueueC7A67616": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "MessageRetentionPeriod": 1209600,
        "QueueName": "dead-letters-alarms-handler-queue-PROD",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
